### PR TITLE
CI Pipeline

### LINKED ISSUE #4: CI/CD Pipeline
Concourse CI PR validation and container release pipelines should be setup for this repository.

### PR DESCRIPTION
**Pull request by ChristopherJMiller**
Added initial pipeline that handles pr validation and image publishing on commits to main.

**TODO**
- ~~Sentry Integration~~

closes #4

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,6 @@
 FROM node:fermium-alpine@sha256:a75f7cc536062f9266f602d49047bc249826581406f8bc5a6605c76f9ed18e98 as builder
 
 WORKDIR /app
-ENV NODE_ENV production
```
Comment (ConnorJC3) [inline Dockerfile:4]: ?

File: src/tools/rpc_client.ts [hunk 2]
```diff
@@ -1,51 +1,41 @@
...
+            default:
                 break;
-            }
         }
     }
 });
 
 stream.on('end', () => {
     console.log('Stream ended.');
 });
```
Comment (ConnorJC3) [inline src/tools/rpc_client.ts:41]: wtf is all this?

File: ci/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+              path: .
+          caches:
+            - path: node_modules
+            - path: .yarn
+          run:
+            path: /bin/sh
+            args:
+              - -exc
+              - |
+                yarn install --cache-folder .yarn
```
Comment (ConnorJC3) [inline ci/ci.yaml:87]: wtf is the point of this?

File: ci/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                echo $VERSION > release/version.txt
+                sentry-cli releases new -p senior-matchmaking "$VERSION"
+                sentry-cli releases set-commits --auto "$VERSION"
+      - load_var: sentry-release
+        file: release/version.txt
+      - task: build
+        privileged: true
+        image: oci-build-task
+        params:
+          BUILD_ARG_RELEASE: ((.:sentry-release))
```
Comment (ConnorJC3) [inline ci/ci.yaml:188]: Unused in Dockerfile?

File: package.json [hunk 5]
```diff
@@ -8,13 +8,12 @@
   "devDependencies": {
```
Comment (ConnorJC3) [inline package.json:8]: package.json changes why?

**Review by ConnorJC3 — CHANGES_REQUESTED** [review]
Completely missing supposed "sentry integration"

File: package.json (hunk 5 above)
Comment (ChristopherJMiller) [reply package.json:8]: Discussed on slack

File: ci/ci.yaml (hunk 4 above)
Comment (ChristopherJMiller) [reply ci/ci.yaml:172]: Fixed in 1dabfd76be8637e7a32c37bdd56c731038f0f0f3

File: ci/ci.yaml (hunk 3 above)
Comment (ChristopherJMiller) [reply ci/ci.yaml]: Fixed in 1dabfd76be8637e7a32c37bdd56c731038f0f0f3

File: src/tools/rpc_client.ts (hunk 2 above)
Comment (ChristopherJMiller) [reply src/tools/rpc_client.ts:41]: @Meachamp did not the run the linter for an unknown amount of time.

File: Dockerfile (hunk 1 above)
Comment (ChristopherJMiller) [reply Dockerfile:4]: Discussed in slack

File: src/server.ts [hunk 6]
```diff
@@ -1,7 +1,27 @@
...
 import { MatchMakingService } from './proto/matchmaking_grpc_pb';
 import { MatchMakingServer } from './mmserver';
 
+global.rootdir = __dirname || process.cwd();
+
+if (process.env.NODE_ENV === 'production') {
+    if (process.env.SENTRY_DSN) {
+        Sentry.init({
+            dsn: process.env.SENTRY_DSN,
+            release: `senior-matchmaking@${process.env.RELEASE}`,
```
Comment (ConnorJC3) [inline src/server.ts:13]: Unnecessary prefix :pepega: ?

File: src/server.ts [hunk 7]
```diff
@@ -1,7 +1,27 @@
 import * as grpc from '@grpc/grpc-js';
+import Sentry from '@sentry/node';
+import { RewriteFrames } from '@sentry/integrations';
 import { MatchMakingService } from './proto/matchmaking_grpc_pb';
 import { MatchMakingServer } from './mmserver';
 
+global.rootdir = __dirname || process.cwd();
+
+if (process.env.NODE_ENV === 'production') {
+    if (process.env.SENTRY_DSN) {
```
Comment (ConnorJC3) [inline src/server.ts:10]: Should probably check for release here too

File: src/server.ts [hunk 8]
```diff
@@ -1,7 +1,27 @@
 import * as grpc from '@grpc/grpc-js';
+import Sentry from '@sentry/node';
+import { RewriteFrames } from '@sentry/integrations';
 import { MatchMakingService } from './proto/matchmaking_grpc_pb';
 import { MatchMakingServer } from './mmserver';
 
+global.rootdir = __dirname || process.cwd();
+
+if (process.env.NODE_ENV === 'production') {
```
Comment (ConnorJC3) [inline src/server.ts:9]: Not necessary because of the following line

**Review by ConnorJC3 — CHANGES_REQUESTED** [review]

**Review by ConnorJC3 — APPROVED** [review]

**Review by Meachamp — APPROVED** [review]
Loogs good

**Pull request merged**